openedx/features/subscriptions/api/v1/tests/test_views.py

```diff
@@ -0,0 +1,157 @@
+""" Commerce API v1 view tests. """
+import itertools
+import json
+from datetime import datetime, timedelta, date
+from random import randint
+
+import ddt
+import pytz
+from django.conf import settings
+from django.contrib.auth.models import Permission
+from django.urls import reverse, reverse_lazy
+from django.utils.http import urlencode
+from django.test.client import Client, RequestFactory
+from django.test import TestCase
+from django.test.utils import override_settings
+
+from openedx.core.djangoapps.site_configuration.tests.factories import SiteFactory
+from openedx.features.edly.tests.factories import (
+    EdlySubOrganizationFactory,
```

Comment:
Can we avoid edly related stuff here?

Reply:
@ziafazal To test we need to make API calls. API calls won't succeed otherwise due to our multisite stuff. One solution that comes to mind is to mock the responses.

Reply:
@ziafazal @rehan99000 since we have our edly access middleware enabled default so it will be trivial to use edly namespace here.

@rehan99000 we can try patching some edly access middleware methods in order to avoid `EdlySubOrganizationFactory` usage. Example from ECOM:

```python
@mock.patch('django.contrib.sites.middleware.get_current_site', mock.Mock(return_value=None))
```

Reply:
secondly, a much more subtle change would be to mock the edly middleware access method. we can remove this in future while pushing it upstream

Reply:
@ziafazal Tests updated. Edly stuff removed